basics/list-comprehension.md

```diff
@@ -0,0 +1,77 @@
+# List Comprehension
+
+
+## Definition
+List comprehensions creates a new list from an existing list that is a shorter syntax than normal. It also processes the list much faster than using a for loop.
+
+
+## Without List Comprehension
+```python
+animals = ["dog", "cat", "giraffe", "donkey", "ape"]
+arr = []
+
+for i in animals:
```

Comment:
Consider improving the variable names, e.g. `arr --> filtered_animals`, `i --> animal`, also in the list comprehension below.